astropy/io/ascii/latex.py

```diff
@@ -97,6 +98,23 @@ def process_val(self, val):
             val = val[1:-1]
         return val
 
+    def gen_process_lines(self, lines):
+        '''Generator function that processes lines and last line specially.'''
+        for x in lines[0:-1]:
```

Comment:
I think slicing a list actually defeats the purpose of a generator. You could use `enumerate` with an `if` to avoid slicing. Or use `itertools.islice`.